include/dynd/exceptions.hpp

```diff
@@ -23,7 +22,7 @@ namespace nd {
   class array;
 } // namespace nd
 
-class DYND_API dynd_exception : public std::exception {
+class DYND_API dynd_exception {
```

Comment:
One suggestion I have here is to add a comment describing why it's not a subclass of std::exception, as that seems like a pretty obvious question to ask when browsing this code.